### PR TITLE
Add the June and July 2023 Steering Council updates.

### PR DESCRIPTION
**Pull request by emilyemorehouse**

### CODE REVIEW

File: updates/2023-07-steering-council-update.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+## 2023-07-24
+
+- The Steering Council met with Łukasz, the Developer-in-Residence, and discussed:
+  - How Europython went (both the conference and the sprints).
+  - dtrace and how to test the probes so they keep functioning.
+  - Miss Islington needs porting to GitHub Actions before we can switch over to requiring 2FA.
+  - The SC asked the DiR to evaluate how we could require code reviews for feature PRs.
+- The SC discussed support for core developer mentorship, namely the budget around a proposal to do an inventory of the current mentorship program.
+- The SC gave our approval to the Security Developer-in-Residence for their CNA proposal.
+- The SC. discussed updating non-borrowing C APIs to match borrowing ones.
```
Comment (JelleZijlstra) [inline updates/2023-07-steering-council-update.md:37]:
```suggestion
- The SC discussed updating non-borrowing C APIs to match borrowing ones.
```

Comment (emilyemorehouse) [reply updates/2023-07-steering-council-update.md]: Thank you!

**Review by brettcannon — APPROVED** [review]

File: updates/2023-06-steering-council-update.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+- The SC decided to schedule [office hours](https://discuss.python.org/t/steering-council-office-hours/28881) a half hour before the weekly meeting, as an experiment.
+- The SC approved [Petr’s proposed changes](https://github.com/python/steering-council/issues/196) to [PEP 387 (Backwards Compatibility Policy)](https://peps.python.org/pep-0387/).
+
+## 2023-06-26
+
+- The Steering Council met with Seth Larson, the new Security Developer-in-Residence. One of Seth’s tasks is to improve the process behind the Python Security Response Team (PSRT), and to investigate the PSRT or PSF becoming a MITRE Sub-CNA, so it has authority over the reporting and ranking of CVEs. Seth will be working mostly with the PSRT team, only planning to have email discussions or ad-hoc meetings with the SC for now.
+- The SC met with Łukasz, the Developer-in-Residence, and discussed:
+  - Łukasz handling the macOS installer build as well as the Windows build for 3.12 beta 4.
+  - His keynote at [PyCon Colombia](https://2023.pycon.co/).
```
Comment (hugovk) [inline updates/2023-06-steering-council-update.md:28]: We can link to the video:
```suggestion
  - His [keynote](https://youtube.com/watch?v=B7U049ll4Rs) at [PyCon Colombia](https://2023.pycon.co/).
```

File: updates/2023-07-steering-council-update.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  - The Deputy Developer-in-Residence job, which has been posted and started receiving applications.
+  - The 3.12 final beta release, which has a few important bug fix PRs that are open.
+  - EuroPython, which is happening next week. Łukasz will be giving a talk and running a panel along with sprints
+- The SC continued to discuss support for core developer mentorship with Deb. The distinction between supporting mentorship for new core developers and improving the process of becoming a core developer was raised – we are specifically looking for how to support existing core developers in mentoring contributors. The SC decided to pause the discussion to solidify what we are looking for in this arrangement.
+- The SC discussed [PEP 703 (Making the Global Interpreter Lock Optional in CPython)](https://peps.python.org/pep-0703/).
+  - The [poll results are in](https://www.notion.so/Weekly-Meeting-Notes-1715f30c2d3d4b8699adb5778e83d695?pvs=21) and indicate that 87% of respondents think we should be actively looking to make Python free-threaded and 63% want to accept and support the maintenance cost of PEP 703.
```
Comment (hugovk) [inline updates/2023-07-steering-council-update.md:15]: I don't have access to notion, is this a better link? 
```suggestion
  - The [poll results are in](https://discuss.python.org/t/poll-feedback-to-the-sc-on-making-cpython-free-threaded-and-pep-703/28540) and indicate that 87% of respondents think we should be actively looking to make Python free-threaded and 63% want to accept and support the maintenance cost of PEP 703.
```

File: updates/2023-07-steering-council-update.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  - Proceeding with caution, the SC decided to formalize a statement on our intention to accept the PEP, along with requirements and expectations for doing so.
+- The SC discussed two C API changes in 3.12 that did not follow PEP-387 to decide how they should be handled, namely:
```
Comment (hugovk) [inline updates/2023-07-steering-council-update.md:17]:
```suggestion
- The SC discussed two C API changes in 3.12 that did not follow [PEP 387 (Backwards Compatibility Policy)](https://peps.python.org/pep-0387/) to decide how they should be handled, namely:
```

File: updates/2023-07-steering-council-update.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+  - Type Object’s tp_dict can now be NULL. This has limited impact and is likely acceptable.
```
Comment (hugovk) [inline updates/2023-07-steering-council-update.md:18]:
```suggestion
  - Type Object’s `tp_dict` can now be `NULL`. This has limited impact and is likely acceptable.
```

File: updates/2023-07-steering-council-update.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+  - PyLongObject implementation and ob_digits. The SC decided to roll back changes only in 3.12, not main (3.13) for now. Overall, this change is okay.
```
Comment (hugovk) [inline updates/2023-07-steering-council-update.md:19]:
```suggestion
  - `PyLongObject` implementation and `ob_digits`. The SC decided to roll back changes only in 3.12, not main (3.13) for now. Overall, this change is okay.
```

File: updates/2023-07-steering-council-update.md [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+## 2023-07-17
+
+- The Steering Council checked in on ongoing conduct issues and the steps to handle them.
+- The SC received feedback from the community on the lack of use for the python-committers email list and decided to close it.
+- The SC continued to discuss [PEP 703 (Making the Global Interpreter Lock Optional in CPython)](https://peps.python.org/pep-0703/), with a focus on establishing a hard list of requirements for acceptance to be included in a forthcoming pre-announcement.
+
+## 2023-07-24
+
+- The Steering Council met with Łukasz, the Developer-in-Residence, and discussed:
+  - How Europython went (both the conference and the sprints).
```
Comment (hugovk) [inline updates/2023-07-steering-council-update.md:31]:
```suggestion
  - How EuroPython went (both the conference and the sprints).
```

**Pull request merged**